apis/network/v1beta1/dnsmasq_types.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// AnnotationHostnameKey -
+	AnnotationHostnameKey = "dnsmasq.network.openstack.org/hostname"
+
+	// DNSDataLabelSelector - label selector to identify config maps with hosts data
+	DNSDataLabelSelector = "dnsmasqhosts"
+)
+
+// DNSMasqOption defines allowed options for dnsmasq
+type DNSMasqOption struct {
+	// +kubebuilder:validation:Enum=server;rev-server;srv-host;txt-record;ptr-record;rebind-domain-ok;naptr-record;cname;host-record;caa-record;dns-rr;auth-zone;synth-domain
+	Key    string   `json:"key"`
+	Values []string `json:"values"`
+}
+
+// DNSMasqSpec defines the desired state of DNSMasq
+type DNSMasqSpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="quay.io/podified-antelope-centos9/openstack-neutron-server:current-podified"
```

Comment:
This should eventually be removed in favor of the env var defaults implemented by webhooks

Reply:
I have added web hook for this. so I think I can remove it

Reply:
done